### PR TITLE
make pyinst.py build name more PEP425 compatible

### PR DESCRIPTION
**Pull request by odo2063**
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

DESCRIPTION

Pyinst.py generates build name "yt-dlp_linux_x86" on ARM(32bit). Also "x86" is not compatible with PEP 425.

Fixes #

Introduces MACHINE variable filled by platform.machine() and replaces "AMD64" with "x86_64" according to PEP 425.

### CODE REVIEW

**Comment by pukkandan** [comment]
The PEP is for wheels... The binary names created by pyinst will not be changed

**Comment by pukkandan** [comment]
Actually, let me clarify. The names for the x86/64 will not be changed. Name change for just the ARM variants are welcome

File: pyinst.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def exe(onedir):
     name = '_'.join(filter(None, (
         'yt-dlp',
         {'win32': '', 'darwin': 'macos'}.get(OS_NAME, OS_NAME),
-        ARCH == '32' and 'x86'
+        ('i' in MACHINE and '86' in MACHINE and 'x86') or (MACHINE != 'x86_64' and MACHINE != 'amd64' and MACHINE)
```
Comment (pukkandan) [inline pyinst.py:68]:
```suggestion
        '' if MACHINE in ('x86_64', 'amd64') else 'x86' if ('i' in MACHINE and '86' in MACHINE) else MACHINE
```
The and syntax is good for short conditions but is unreadable when it's this complex

**Comment by pukkandan** [comment]
There are other changes needed in L68, L125 etc. Pls look into the full scope of the changes that you are making and not just your usecase

**Comment by odo2063** [comment]
Sorry, we were not at L125. Now we've learned to read full file before pull requesting.

File: pyinst.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def exe(onedir):
     name = '_'.join(filter(None, (
         'yt-dlp',
         {'win32': '', 'darwin': 'macos'}.get(OS_NAME, OS_NAME),
-        ARCH == '32' and 'x86'
+        '' if MACHINE in ('x86_64', 'amd64') else 'x86' if ('i' in MACHINE and '86' in MACHINE) else MACHINE
```
Comment (pukkandan) [inline pyinst.py:68]: I think it makes sense to put this in a function/global var so that it can be reused below

Comment (odo2063) [reply pyinst.py:68]: I agree, if you allow I'll prepare something tomorow

**Comment by odo2063** [comment]
I hope that fits your quality standards better

**Comment by pukkandan** [comment]
Wish you'd have actually tested the code when making a PR! https://github.com/yt-dlp/yt-dlp/commit/d08e1e68758d5041afa79abd6a2d7dd1c45879d8

**Pull request merged**